core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/part/EnvelopePart.java

```diff
@@ -616,6 +613,65 @@ public EnvelopePart copyAndShift(double positionDelta, double minPosition, doubl
                 new HashMap<>(attrs), newPositions.toArray(), newSpeeds.toArray(), newTimeDeltas.toArray());
     }
 
+    /**
+     * Compute the minimum EnvelopePart between 2 envelope parts with an intersecting range [a, b].
+     * The resulting envelope part starts at the envelope with the minimum speed at a and ends at b.
+     */
+    public static EnvelopePart min(
+            EnvelopePart envelopePartA, EnvelopePart envelopePartB, Iterable<SelfTypeHolder> attrs) {
```

Comment:
Side note: we *could* just iterate meter per meter, interpolate each envelope, and pick the min (and *maybe* simplify it). It would probably be slower, but the difference may not be important enough to be an issue.

Though this method with the key position doesn't add that much complexity either.

Reply:
No it's fine as it is